Docs: AWS RDS Proxy Guide

greedy52 opened the pull request:
Closes https://github.com/gravitational/teleport/issues/18047

Thanks @alexfornuto for the initial [draft](https://github.com/gravitational/teleport/pull/21234).

Commit:
init RDS proxy guide

greedy52 committed:
update diagram

greedy52 committed:
update doc

greedy52 committed:
add missing png

alexfornuto commented on docs/pages/database-access/guides/rds-proxy.mdx:
```diff
@@ -0,0 +1,228 @@
+            "Effect": "Allow",
+            "Action": "secretsmanager:GetSecretValue",
+            "Resource": [
+                "arn:aws:secretsmanager:us-west-1:account_id:secret:rdsproxy_alice",
```
Just flagging this as another reason to prioritize https://github.com/gravitational/docs/issues/186, as we could replace `alice` with a Var everywhere else but here.

Commit:
copy edits...

Minor phrasing adjustment, replaced scoped blocks for cloud with Vars, fixed code fencing for commands and json

alexfornuto reviewed (approved):
Thanks @greedy52! In the interest of time I made my copy edits as a commit, but of course nothing is set in stone when it comes to git.

ptgott commented on docs/pages/database-access/guides/rds-proxy.mdx:
```diff
@@ -0,0 +1,201 @@
+description: How to configure Teleport Database Access with AWS RDS Proxy.
+---
+
+This guide will help you to set up Teleport to access your AWS RDS proxies.
```
```suggestion
This guide will help you to set up Teleport to access your AWS RDS Proxy instances.
```
It looks like the AWS docs say "RDS Proxy instance", instead of "RDS Proxies" (https://aws.amazon.com/rds/proxy/)

ptgott commented on docs/pages/database-access/guides/rds-proxy.mdx:
```diff
@@ -0,0 +1,201 @@
+</ScopedBlock>
+
+<Admonition type="note" title="Supported Engine Family">
+Teleport currently supports RDS Proxies with engine family **PostgreSQL** or **MariaDB and MySQL**.
```
I think we should `s/RDS (p|P)roxies/RDS Proxy instances/` throughout.

ptgott commented on docs/pages/database-access/guides/rds-proxy.mdx:
```diff
@@ -0,0 +1,201 @@
+
+## Step 4/5. Configure database user credentials
+
+The Database Service connects to a RDS Proxy instance using IAM authentication.
```
```suggestion
The Database Service connects to an RDS Proxy instance using IAM authentication.
```

ptgott commented on docs/pages/database-access/reference/aws.mdx:
```diff
@@ -153,6 +153,77 @@ policies for each discovery type are shown below.
+
+<Tabs>
+  <TabItem label="IAM user">
+  Use this policy if you're connecting to RDS Proxies and your Teleport
```
```suggestion
  Use this policy if you're connecting to RDS Proxy instances and your Teleport
```

ptgott commented on docs/pages/database-access/reference/aws.mdx:
```diff
@@ -153,6 +153,77 @@ policies for each discovery type are shown below.
+<Tabs>
+  <TabItem label="IAM user">
+  Use this policy if you're connecting to RDS Proxies and your Teleport
+  database agent runs as an IAM user (for example, uses an AWS credentials
```
```suggestion
  Database Service instance runs as an IAM user (for example, uses an AWS credentials
```

ptgott reviewed (approved):
Approved with suggestions

greedy52 committed:
RDS Proxies -> RDS Proxy instances

greedy52 committed:
database agent -> Database Service instance
